### ISSUE TITLE
Add Lscale variable as an intent(out) to advance_clubb_core

### ISSUE THREAD

**Issue by adamrher** [larson-group/clubb_release#16]
Requesting the Lscale variable be added as an intent(out) argument to advance_clubb_core, and its api (called from clubb_intr.F90 in CAM). The purpose is for modifying the khzm diffusion coefficient computed in clubb with additional kinetic energy terms stemming from the CLUBB+MF CPT. In CAM, khzm is used outside of clubb to transport aerosols.

I also had trouble accessing the ```c_K``` variable used for computing khzm. In advance_clubb_core:
```
    !----------------------------------------------------------------
    ! Eddy diffusivity coefficient
    !----------------------------------------------------------------
    ! c_K is 0.548 usually (Duynkerke and Driedonks 1987)
    ! CLUBB uses a smaller value to better fit empirical data.

    ! Calculate CLUBB's eddy diffusivity as
    !   CLUBB's length scale times a velocity scale.
    c_K = clubb_params(ic_K)

    !$acc parallel loop gang vector collapse(2) default(present)
    do k = 1, nz
      do i = 1, ngrdcol
        Kh_zt(i,k) = c_K * Lscale(i,k) * sqrt_em_zt(i,k)
      end do
    end do
```

If this could also be made available in the clubb_intr.F90 interface that would be preferable to what I'm currently doing (duplicating and hard coding this value).

**Comment by vlarson** [comment]
> Requesting the Lscale variable be added as an intent(out) argument to advance_clubb_core, and its api (called from clubb_intr.F90 in CAM). The purpose is for modifying the khzm diffusion coefficient computed in clubb with additional kinetic energy terms stemming from the CLUBB+MF CPT. In CAM, khzm is used outside of clubb to transport aerosols.

OK, this sounds doable.

> I also had trouble accessing the `c_K` variable used for computing khzm. In advance_clubb_core: 
> If this could also be made available in the clubb_intr.F90 interface that would be preferable to what I'm currently doing (duplicating and hard coding this value).

In clubb_intr, `c_K` should be [brought into `clubb_params_single_col`](https://github.com/ESCOMP/CAM/blob/eb275095c52e54b00c55191b6ab7bca614004a57/src/physics/cam/clubb_intr.F90#L1662) and later [into `clubb_params`](https://github.com/ESCOMP/CAM/blob/eb275095c52e54b00c55191b6ab7bca614004a57/src/physics/cam/clubb_intr.F90#L3162).   Hence `c_K` should be accessible in clubb_intr by a line like `c_K = clubb_params(i,ic_K)`.  Is this good enough for `c_K`?

**Comment by adamrher** [comment]
Yes, I was able to access ```c_K``` by invoking ```clubb_params(ic_K)```. That's good to know, thanks. And so I'm just requesting the Lscale from the api.

**Comment by vlarson** [comment]
Lscale was introduced as an output variable in the argument list of advance_clubb_core_api in [this commit](https://github.com/larson-group/clubb_release/commit/91a850ea46938fd79250eaada0aef3e4aa5ad30e). @adamrher, please let me know if you need any further updates.
